File: lib/models/data/pos_host_details.dart
import 'detail_item.dart';

class PosHostDetails {
  final DetailItem credit;
  final DetailItem debit;
  final DetailItem total;

  PosHostDetails({
    required this.credit,
    required this.debit,
    required this.total,
  });

  factory PosHostDetails.fromJson(Map<String, dynamic> json) {
    return PosHostDetails(
      credit: DetailItem.fromJson(json['credit'] as Map<String, dynamic>),
      debit: DetailItem.fromJson(json['debit'] as Map<String, dynamic>),
      total: DetailItem.fromJson(json['total'] as Map<String, dynamic>),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'credit': credit.toJson(),
      'debit': debit.toJson(),
      'total': total.toJson(),
    };
  }
}

File: android/terminalSdk/src/main/java/io/nearpay/terminalsdk/data/dto/PosHostDetails.kt
package io.nearpay.terminalsdk.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


@Serializable
data class PosHostDetails(
    @SerialName("credit")
    val credit: DetailItem,

    @SerialName("debit")
    val debit: DetailItem,

    @SerialName("total")
    val total: DetailItem
)